tests/test_cli_compile.py

```diff
@@ -1063,6 +1140,40 @@ def test_annotate_option(pip_conf, runner, option, expected):
     assert out.exit_code == 0
 
 
+@pytest.mark.parametrize(
+    ("input_filename", "input_content"),
+    (
+        ("requirements.in", "small-fake-a==0.1"),
+        (
+            "setup.py",
+            (
+                "from setuptools import setup\n"
+                "setup(name='fake-setuptools-a', install_requires=['small-fake-a==0.1'])"
+            ),
+        ),
+    ),
```

Comment:
It's hard to select a test to run with an identifier like `test_annotation_relative_paths[setup.py-from setuptools import setup\nsetup(name='fake-setuptools-a', install_requires=['small-fake-a==0.1'])]`. When the params are complex, it's best to set their IDs explicitly:
```suggestion
    ),
    ids=('requirements.in', 'setup.py'),
```

Reply:
For history, here's how the command looks:
```console
$ tox -e py -- tests/test_cli_compile.py::"test_annotation_relative_paths[setup.py-from setuptools import setup\\nsetup(name='fake-setuptools-a', install_requires=['small-fake-a==0.1'])]"
```

Reply:
Ah, thanks, I'll add the IDs as suggested. I'd been calling the test with, e.g. `pytest -k annotation_rel` and not picking out a particular parameter set.

Reply:
FYI the best practice is to accept the suggested changes via GH UI because this preserves the authorship of the respective contributions.

Reply:
Also, IDs are useful beyond invocation: the test report looks nicer this way too because the context of params is not really implied in the complex structures.

Reply:
dbad136ec1f8e49cb947818e6328bb8962f93d28

Reply:
> FYI the best practice is to accept the suggested changes via GH UI because this preserves the authorship of the respective contributions.

What you put here in the UI wouldn't have passed the QA checks, so I just set you as co-author, but I can amend the commit to remove myself, sorry.

Reply:
20e809ab0add01e79a822a00cb0946af75ec77ff

Reply:
> What you put here in the UI wouldn't have passed the QA checks, so I had just made the change and set you as co-author, but I can amend the commit to remove myself, sorry.

Nah.. That's alright! That's actually exactly what GH does — it adds a `Co-Authored-By` while the Committer/Author fields still show the person who pushed the button. My workflow is normally to accept the changes, pull locally, and then amend things if necessary. This is because it's the easiest way to discover the suggester's email w/o having to poke GH API for some old commits if it's hidden.

Reply:
By the way, being an org `Member` you should be able to edit other people's comments so you can adjust the suggestions before hitting "commit".

Reply:
> This is because it's the easiest way to discover the suggester's email w/o having to poke GH API for some old commits if it's hidden.

LOL yeah I cloned one of your random repos to set the co-author, before I removed myself as an author.

Thanks maybe I'll edit it in here next time, I just don't like committing in the browser, I'd usually rather manually set the co-author and use my native tools.

Reply:
OK it looks like I can at least get setuptools installed by *removing* the `pip_conf` fixture from the call. I don't yet know for *sure* if it's getting setuptools from the network.

FWIW I'm going ahead this way to push a commit with passing tests soon, while remaining very open to improvements.

Reply:
> LOL yeah I cloned one of your random repos to set the co-author, before I removed myself as an author.

Oh, you could just open something like https://api.github.com/repos/AndydeCleyre/bookme/commits in your browser and `Ctrl+F` `@`

Reply:
But I think the problem here is that GitHub knows if somebody prefers a hidden/private noreply email that GH sets for the users or one of the tens of their emails that are per-project. And this is another reason I use this method since I don't want to guess if the other person wanted that specific email in this specific Git tree...

Reply:
> OK it looks like I can at least get setuptools installed by _removing_ the `pip_conf` fixture from the call. I don't yet know for _sure_ if it's getting setuptools from the network.

You could as well inject the underlying `make_pip_conf` fixture and make a custom config if necessary:
```conf
    make_pip_conf(
        dedent(
            f"""\
            [global]
            find-links = {MINIMAL_WHEELS_PATH}
            """
        )
    )
```